homeassistant/components/deconz/__init__.py

```diff
@@ -33,8 +34,11 @@ async def async_setup_entry(hass, config_entry):
 
     gateway = DeconzGateway(hass, config_entry)
 
-    if not await gateway.async_setup():
-        return False
+    try:
+        if not await gateway.async_setup():
+            return False
+    except Exception as error:
+        raise ConfigEntryNotReady from error
```

Comment:
should this just be a raise to pass the already thrown exception directly since there is a difference between an authorization failed exception (dont retry) and a failed to connect to gateway (which already raises a ConfigEntryNotReady in gateway.py)?